### PR TITLE
Changed right management system to something closer to SF2's security

### PR DESCRIPTION
**Pull request by jtabet**
| Questions | Answers |
| --- | --- |
| Branch? | Develop |
| Description? | I added an Access class and tried to remove logic duplicates as much as possible. I also changed the access and module_access structure to use roles with slug. **It still needs improvements** like merging the access table and module_access table into one, adding tree support for roles and improving the Symfony part. |
| Type? | Improvement |
| Category? | BO |
| BC breaks? | Yes |
| Deprecations? | No |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-393 |
| How to test? | Change access settings in the "employees/permissions" page and check on the page if the newly defined permissions work correctly |

### CODE REVIEW

File: classes/Access.php [hunk 1]
```diff
@@ -0,0 +1,351 @@
+<?php
+/**
+ * 2007-2015 PrestaShop
```
Comment (mickaelandrieu) [inline classes/Access.php]: 2016*

**Comment by mickaelandrieu** [comment]
Nice job, let's take time to review and test it :)

**Comment by jtabet** [comment]
@dSkrbic

**Comment by dSkrbic** [comment]
Hello,

Tested, and **OK**

_See bug reports for comments_

File: admin-dev/themes/default/template/controllers/access/helpers/form/form.tpl [hunk 2]
```diff
@@ -23,393 +23,393 @@
...
-			var tout = $(this).data('rel').split('||');
-			var id_tab = tout[0];
-			var id_profile = tout[1];
-			var perm = tout[2];
-			var enabled = $(this).is(':checked')? 1 : 0;
-			var tabsize = tout[3];
-			var tabnumber = tout[4];
-			var table = 'table#table_'+id_profile;
+      $('.ajaxPower').change(function(){
+         var tout = $(this).data('rel').split('||');
```
Comment (mickaelandrieu) [inline admin-dev/themes/default/template/controllers/access/helpers/form/form.tpl]: ``tout``?

File: admin-dev/themes/default/template/controllers/access/helpers/form/form.tpl [hunk 3]
```diff
@@ -23,393 +23,393 @@
...
-		$(".changeModuleAccess").change(function(){
-			var tout = $(this).data('rel').split('||');
-			var id_module = tout[0];
-			var perm = tout[1];
-			var id_profile = tout[2];
-			var enabled = $(this).is(':checked') ? 1 : 0;
-			var enabled_attr = $(this).is(':checked') ? true : false;
-			var table = 'table#table_module_'+id_profile;
+      $(".changeModuleAccess").change(function(){
+         var tout = $(this).data('rel').split('||');
```
Comment (mickaelandrieu) [inline admin-dev/themes/default/template/controllers/access/helpers/form/form.tpl]: same here, is it a french var name ?

**Pull request merged**